### ISSUE TITLE
Accessibility issues with "Create New Document" page

### ISSUE THREAD

**Issue by dariiing** [cryptpad/cryptpad#1912]
### Description

This page is currently not accessible for keyboard or screen reader users. 

Issues:

- keyboard traps 
- non-interactive elements are receiving focus
- focus outlines are missing
- checkboxes are not announced correctly by screen readers (checked/unchecked state)
- some elements lack accessible labels and are not announced properly
- interactive elements do not receive focus 
- elements are not announced
![Image](https://github.com/user-attachments/assets/6caf5357-6c75-472d-8772-319d0ed72ef3)

### Steps to reproduce

1. Create a new document
2. Try tabbing through elements
3. Use screen reader to interact

### Relevant logs

```shell

```

### Expected behavior

Interactive elements should be reachable via keyboard, and all elements should be announced by screen readers properly

### Which operating system are you using?

Linux/BSD/UNIX

### What browsers are you seeing the problem on?

Mozilla Firefox

### Extensions

.

### Version

2025.3.0
